orangecontrib/imageanalytics/widgets/owimageviewer.py

```diff
@@ -1190,6 +1190,16 @@ def __init__(self, parent=None):
             )
             netmanager.setCache(cache)
             ImageLoader._NETMANAGER_REF = weakref.ref(netmanager)
+
+            proxy_address = os.environ.get('http_proxy')
+            if proxy_address is not None:
+                ip_port = proxy_address.split(":")
+                ip, port = ip_port \
+                    if len(ip_port) == 2 else (ip_port[0], '8080')
+                proxy = QNetworkProxy(
+                    QNetworkProxy.HttpProxy, ip, int(port))
+                netmanager.setProxy(proxy)
+
```

Comment:
Maybe try this:
```diff
@@ -31,7 +31,8 @@ from AnyQt.QtCore import (
 )
 from AnyQt.QtCore import pyqtSignal as Signal, pyqtSlot as Slot
 from AnyQt.QtNetwork import (
-    QNetworkAccessManager, QNetworkDiskCache, QNetworkRequest, QNetworkReply
+    QNetworkAccessManager, QNetworkDiskCache, QNetworkRequest, QNetworkReply,
+    QNetworkProxyFactory, QNetworkProxyQuery
 )
 
 import Orange.data
@@ -1170,6 +1171,11 @@ class OWImageViewer(widget.OWWidget):
         self.clear()
 
 
+class SystemProxyFactory(QNetworkProxyFactory):
+    def queryProxy(self, query=QNetworkProxyQuery()):
+        return self.systemProxyForQuery(query)
+
+
 class ImageLoader(QObject):
     #: A weakref to a QNetworkAccessManager used for image retrieval.
     #: (we can only have one QNetworkDiskCache opened on the same
@@ -1189,6 +1195,7 @@ class ImageLoader(QObject):
                              __name__ + ".ImageLoader.Cache")
             )
             netmanager.setCache(cache)
+            netmanager.setProxyFactory(SystemProxyFactory())
             ImageLoader._NETMANAGER_REF = weakref.ref(netmanager)
         self._netmanager = netmanager
```

At the least do not parse http_proxy like that. For instance that fails for ```http://user:1234@example.com:8000``` Use `QUrl` or `urlparse`.

Reply:
@ales-erjavec thank you for the suggestion. I tried this implementation on our WIFI which uses the proxy (Orange 5G, proxy IP is: 192.168.0.1:3128), but cannot make it work. Do you have an idea what to try or how to debug it?

Reply:
Works for me on macos 10.11 with Qt 5.9.3 and system configured proxy.

Can you try https://github.com/ales-erjavec/orange3-imageanalytics/tree/proxy and configure via the orange preferences dialog.

Reply:
Work for me too #94